spring-pulsar/src/main/java/org/springframework/pulsar/listener/GenericPulsarListenerContainerRegistry.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.listener;
+
+import java.util.Collection;
+import java.util.Set;
+
+import org.springframework.lang.Nullable;
+
+/**
+ * A registry for containers.
+ *
+ * @param <T> Message payload type.
+ * @author Christophe Bornet
+ */
+public interface GenericPulsarListenerContainerRegistry<T> {
+
+	@Nullable
+	T getListenerContainer(String id);
```

Comment:
Do you mind adding javadoc here describing these methods? Also, what is the diff between "getListenerContainers" and "getAllListenerContainers"?

Reply:
Done